### ISSUE TITLE
Browser ONLY support for Broadcast votes?

### ISSUE THREAD

**Issue by unipsycho** [adcpm/steem#43]
So, your broadcast example is listed along with all the browser only requests which I have the requests working well in my javascript app.  However, I can't use the broadcast command:

> var steem = require('steem');
> 
> var wif = steem.auth.toWif(username, password, 'posting');
> steem.broadcast.vote(wif, voter, author, permlink, weight, function(err, result) {
>     console.log(err, result);
> });

I just get the following error.  Which looks like a timeout, but not really sure what might be the problem.  Are these broadcasts able to run in the browser in JS directly with SteemJS or is it requiring server side with node.js or something?  This isn't clear to me, since everything else is client side?  Any help would be great.

> Error: 10 assert_exception: Assert Exception
> now < trx.expiration: 
>     {"now":"2016-12-13T06:44:18","trx.exp":"2016-12-13T06:43:52"}
>     th_a  database.cpp:3267 _apply_transaction

Thanks.

**Comment by adcpm** [comment]
You can rather use Steem.js on browser or service side, you don't need server side to broadcast vote if you don't want to. The vote should work, this error come from the blockchain, my guess is that you are voting for a post too old. Can you show me the params you are using?

**Comment by svk31** [comment]
The error message means that the expiration time of the transaction was already passed by the time it was broadcas,t, so it was rejected by the blockchain. Could be an issue with your local clock being out of sync for example.

**Comment by unipsycho** [comment]
Thanks for the ideas guys.  This is using on recent posts, so don't think it too old.  My clock is in sync with an online time service so not sure how that could cause it either.  Either way, Here is my function for this vote, still get the same error response.

> $scope.vote = function() {
>     var wif = steem.auth.toWif('unipsycho', PRIVATE_POSTING_KEY, 'posting');
>     steem.broadcast.vote(PRIVATE_POSTING_KEY, 'unipsycho', 'countryinspired', 'steemit-photo-challenge-entry-1-reflections', 10000, function(err, result) {
>         console.log(err, result);
>     });
> }

**Comment by adcpm** [comment]
Can you check on your console network when the vote is broadcasted if the "expiration" time is UTC?
![image](https://cloud.githubusercontent.com/assets/16245250/21160308/b40d55ec-c184-11e6-8ce2-cfd0478904a8.png)

**Comment by unipsycho** [comment]
Looks like the expiration time is in UTC, yes, but the now going out is still before the expiration.  I set my local clock back a couple minutes and tried in UTC as well, but the now (shown in the console) is still going out a couple minutes before the expiration for some reason.
![image](https://cloud.githubusercontent.com/assets/12666072/21162718/c5c6b466-c186-11e6-9c5d-1235b2949440.png)

**Comment by unipsycho** [comment]
OK, so playing around more with this, I set to use a time server to auto sync my clock and timezone then I tried again, got a voting power error instead.  I dropped the voting power and voila, the vote went through, so not sure why the clock is so critical to this, but its working with this, so thanks for the help guys in figuring this out, I hope it gives you a closer idea what went wrong as well.  now I can get back to coding.  haha
